Add utility function make_tensor

devashisht opened the pull request:
Summary: Current implementation of Tensor takes an intrusive_ptr as an argument for storing data. But instead of requiring users to explicitly pass an intrusive_ptr we want them to pass args for intrusive ptr directly which are forwarded internally through new helper function called make_tensor

Differential Revision: D10152661

ezyang commented:
cc @gchanan

apaszke commented on aten/src/ATen/core/Tensor.h:
```diff
@@ -677,6 +677,15 @@ struct CAFFE2_API WeakTensor {
+// in an intrusive_ptr instead it just converts the argument passed to
+// requested intrusive_ptr type.
+template <typename T, typename... Args>
+Tensor make_tensor(Args&&... args) {
```
Is this ever supposed to be used outside of the library? If no, can we put it in a `detail` namespace? Otherwise it might end up being confusing for people

devashisht committed:
Add utility function make_tensor (#12288)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/12288

Current implementation of Tensor takes an intrusive_ptr as an argument for storing data. But instead of requiring users to explicitly pass an intrusive_ptr we want them to pass args for intrusive ptr directly which are forwarded internally through new helper function called make_tensor

Reviewed By: ezyang

Differential Revision: D10152661

fbshipit-source-id: 8dbf73a552c8ccc26918005b211d385dd4030f8c
